src/sql/workbench/services/notebook/browser/sql/sqlSessionManager.ts

```diff
@@ -327,7 +327,9 @@ class SqlKernel extends Disposable implements nb.IKernel {
 				// TODO when we can just show error as an output, should show an "execution canceled" error in output
 				this._future.handleDone().catch(err => onUnexpectedError(err));
 			}
-			this._queryRunner.runQuery(code).catch(err => onUnexpectedError(err));
+			this._connectionManagementService.refreshAzureAccountTokenIfNecessary(this._connectionPath).then(() => {
+				this._queryRunner.runQuery(code).catch(err => onUnexpectedError(err));
```

Comment:
Why not have the query runner refresh the token? That'd hit a lot more places than just this specific one.